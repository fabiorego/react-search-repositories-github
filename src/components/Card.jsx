import React from 'react';

export default function Card(props) {
  function onClickView() {
    window.open(`${props.url_repo}`, '_blank');
  }

  return (
    <div class="container">
      <div id="repos-list" class="column">
        <div class="col-md-4 mb-4">
          <div className="">
            <div className="">
              <img src={props.url_avatar} className="my-img-profile" alt="Avatar Usuario" />
            </div>

            <div className="">
              <div className="">{props.name}</div>
              <div className="">
                by <b>{props.username}</b>
              </div>
            </div>

            <div className="">{props.description}</div>

            <div className="">
              <button onClick={onClickView} className="">
                View
              </button>
              <button className="">
                <i className="fas fa-star"></i> {props.stars}
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
