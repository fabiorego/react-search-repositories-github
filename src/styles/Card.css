.card-result {
  width: 30%;
  height: 100%;
  margin-bottom: 10px;
  margin-left: 15px;
  margin-right: 15px;
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  flex-wrap: wrap;
  background-color: #ffffff;
  border: 0.5px solid rgba(1, 1, 1, 0.1);
  border-bottom: 0;
  border-radius: 3px;

  -webkit-box-shadow: 0px 0px 5px 0px rgba(240, 240, 240, 1);
  -moz-box-shadow: 0px 0px 5px 0px rgba(240, 240, 240, 1);
  box-shadow: 0px 0px 5px 0px rgba(240, 240, 240, 1);

  cursor: pointer;
}
.card-result-img {
  width: 25%;
  height: 100px;
  padding: 10px 10px 10px 10px;
  display: flex;
  align-items: center;
  justify-content: center;
}
.card-result-img-profile {
  width: 60px;
  border-radius: 40%;
}
.card-result-repo {
  width: 75%;
  height: 70px;
  padding-left: 10px;
}
.card-result-repo-name {
  width: 100%;
  height: 40px;
  padding-top: 12px;
  display: flex;
  align-items: flex-end;
  font-weight: 400;
  font-size: 14px;
}
.card-result-repo-username {
  width: 100%;
  height: 40px;
  font-size: 14px;
}
.card-result-repo-description {
  width: 100%;
  min-height: 30px;
  padding: 10px 20px 20px 20px;
  font-size: 14px;

  color: rgba(1, 1, 1, 0.6);
}
.card-result-repo-actions-buttons {
  width: 100%;
  min-height: 20px;
}
.card-result-repo-buttons {
  width: 25%;
  padding: 4px;

  font-size: 14px;
  font-weight: bold;
  color: rgba(1, 1, 1, 0.6);
  background-color: rgba(115, 115, 115, 0.1);

  border: 1px solid rgba(1, 1, 1, 0.1);

  outline: none;
  text-decoration: none;
}
.card-result-repo-button-view {
  border-radius: 0px 0px 0px 3px;
  border-right: 0;
  border-left: 0;
  background-color: #f8f8ff;
}
.card-result-repo-button-forks {
  border-radius: 0px 0px 3px 0px;
  background-color: #f8f8ff;
}
.card-result-repo-button-watchers {
  border-radius: 0px 0px 3px 0px;
  background-color: #f8f8ff;
}
.card-result-repo-button-stars {
  border-radius: 0px 0px 3px 0px;
  background-color: #f8f8ff;
}

@media (max-width: 1039.98px) {
  .card-result {
    width: 100%;
  }
}
