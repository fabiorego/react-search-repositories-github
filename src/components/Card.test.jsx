import React from 'react';
import Card from './Card';
import renderer from 'react-test-renderer';

let findTextElement = function(tree, element) {
  return true;
};

test('Card snapshot', () => {
  const component = renderer.create(<Card>Card</Card>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Find View element', () => {
  let tree = renderer.create(<Card />).toJSON();
  expect(findTextElement(tree, 'View')).toBeDefined();
});
