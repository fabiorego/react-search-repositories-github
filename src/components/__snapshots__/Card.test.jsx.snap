// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot 1`] = `
<div
  className="card-result"
>
  <div
    className="card-result-img"
  >
    <img
      alt="Avatar Usuario"
      className="card-result-img-profile"
    />
  </div>
  <div
    className="card-result-repo"
  >
    <div
      className="card-result-repo-name"
    />
    <div
      className="card-result-repo-username"
    >
      <b />
    </div>
  </div>
  <div
    className="card-result-repo-description"
  />
  <div
    className="card-result-repo-actions-buttons"
  >
    <button
      className="card-result-repo-buttons card-result-repo-button-view"
      onClick={[Function]}
    >
      View
    </button>
    <button
      className="card-result-repo-buttons card-result-repo-button-watchers"
    >
      <i
        className="fas fa-eye"
      />
       
    </button>
    <button
      className="card-result-repo-buttons card-result-repo-button-stars"
    >
      <i
        className="fas fa-star"
      />
       
    </button>
    <button
      className="card-result-repo-buttons card-result-repo-button-forks"
    >
      <i
        className="fas fa-code-branch"
      />
       
    </button>
  </div>
</div>
`;
